infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SshKeysProvisioner.java

```diff
@@ -114,19 +135,28 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
                     identity.getOwnerId(), "vcs", "default-" + new Date().getTime()));
       } catch (ServerException | ConflictException e) {
         LOG.warn("Unable to generate the initial SSH key. Cause: {}", e.getMessage());
-        return;
+        return emptyList();
       }
     }
+    return sshPairs;
+  }
 
-    doProvisionSshKeys(sshPairs, k8sEnv, identity.getWorkspaceId());
-
-    StringBuilder sshConfigData = new StringBuilder();
-    for (SshPair sshPair : sshPairs) {
-      sshConfigData.append(buildConfig(sshPair.getName()));
+  /**
+   * Return system (aka 'internal') ssh keys, this key used for internal services like rsync via ssh
+   * and etc. Not generated and managed by a user
+   *
+   * @param identity
+   * @return list of keys pair
+   */
+  private List<SshPairImpl> getSystemSshPairs(RuntimeIdentity identity) {
```

Comment:
Maybe it would be better to be more explicit here and call it `getRsyncSshPair()`? IMO `SystemSshPairs` is pretty generic and might be misleading

Reply:
I think generic names here is acceptable, because it can be used for different services in future

Reply:
could you provide an example? I'm ok to have a generic names, but at the same time if we would have the clear naming from the begging this wil make the code more readable / grokable

Reply:
Hard to say about real world example, but I that come to the mind: some service for SSH Tunneling or SCP, for example